actix-router/src/resource.rs

```diff
@@ -254,38 +489,47 @@ impl ResourceDef {
         }
     }
 
-    /// Is the given path and parameters a match against this pattern.
-    pub fn match_path<T: ResourcePath>(&self, path: &mut Path<T>) -> bool {
-        profile_method!(match_path);
-        self.match_path_checked(path, &|_, _| true, &Some(()))
+    /// Collects dynamic segment values into given path.
+    ///
+    /// Returns `true` if `path` matches this resource.
+    pub fn capture_match_info<T: ResourcePath>(&self, path: &mut Path<T>) -> bool {
+        profile_method!(is_path_match);
+        self.capture_match_info_fn(path, &|_, _| true, &None::<()>)
     }
 
-    /// Is the given path and parameters a match against this pattern?
-    pub fn match_path_checked<R, T, F, U>(
+    /// Collects dynamic segment values into given resource after matching paths and executing
+    /// check function.
+    ///
+    /// The check function is given a reference to the passed resource and optional arbitrary data.
+    ///
+    /// Returns `true` if resource path matches this resource definition using the supplied check function.
+    ///
+    /// The check function is supplied with the resource `res` and `user_data`.
+    pub fn capture_match_info_fn<R, T, F, U>(
         &self,
         res: &mut R,
-        check: &F,
+        check_fn: &F,
         user_data: &Option<U>,
     ) -> bool
     where
-        T: ResourcePath,
         R: Resource<T>,
+        T: ResourcePath,
         F: Fn(&R, &Option<U>) -> bool,
     {
-        profile_method!(match_path_checked);
+        profile_method!(is_path_match_fn);
 
-        let mut segments: [PathItem; MAX_DYNAMIC_SEGMENTS] = Default::default();
+        let mut segments = <[PathItem; MAX_DYNAMIC_SEGMENTS]>::default();
         let path = res.resource_path();
 
-        let (matched_len, matched_vars) = match self.pat_type {
+        let (matched_len, matched_vars, tail) = match self.pat_type {
             PatternType::Static(ref segment) => {
```

Comment:
```suggestion
        let (matched_len, matched_vars, tail) = match self.pat_type {
            PatternType::Static(_) | PatternType::Prefix(_) => match self.find_match(path.path()) {
                Some(len) => (len, None, None),
                None => return false, 
            }
```

at the cost of one call.